### PR TITLE
Docs: Add Securing BlueChi with mTLS and double proxy

### PR DESCRIPTION
**Pull request by ygalblum**

### CODE REVIEW

File: subprojects/hashmap.c
Comment (mwperina) [inline subprojects/hashmap.c:1]: Please remove those changes from the PR

Comment (engelmi) [reply subprojects/hashmap.c:1]: I think a `git submodule update` should be sufficient. This is probably also the reason the unit tests and rpmbuild fail.

File: doc/docs/img/bluechi_secure_proxy_connection.png
Comment (engelmi) [inline doc/docs/img/bluechi_secure_proxy_connection.png:1]: Could you add those diagrams to the [diagrams.drawio.xml](https://github.com/eclipse-bluechi/bluechi/blob/main/doc/diagrams.drawio.xml), e.g. by opening it in the online editor (https://app.diagrams.net) and adding for each image a new tab? This way when we have to update/recreate those diagrams we can easily do so.

File: subprojects/hashmap.c
Comment (ygalblum) [reply subprojects/hashmap.c:1]: Done

File: doc/docs/img/bluechi_secure_proxy_connection.png
Comment (ygalblum) [reply doc/docs/img/bluechi_secure_proxy_connection.png:1]: Done. I hope this is what you meant :)

Comment (engelmi) [reply doc/docs/img/bluechi_secure_proxy_connection.png:1]: Yes, looks good :) Thanks!

**Review by engelmi — APPROVED** [review]
LGTM

**Pull request merged**